platform-api/che-core-api-factory/src/test/java/org/eclipse/che/api/factory/server/FactoryServiceTest.java

```diff
@@ -992,6 +958,117 @@ public void shoutFindByAttribute() throws Exception {
     }
 
 
+    @Test
+    public void shouldGenerateFactoryJsonIncludeAllProjects() throws Exception {
+        // given
+        final String wsId = "workspace123234";
+        UsersWorkspaceImpl.UsersWorkspaceImplBuilder ws = UsersWorkspaceImpl.builder();
+        ws.setId(wsId);
+        ws.setProjects(Arrays.asList(dto.createDto(ProjectConfigDto.class)
+                                        .withSource(
+                                                dto.createDto(SourceStorageDto.class)
+                                                   .withType("git")
+                                                   .withLocation("location")),
+                                     dto.createDto(ProjectConfigDto.class)
+                                        .withSource(dto.createDto(SourceStorageDto.class)
+                                                       .withType("git")
+                                                       .withLocation("location"))));
+        ws.setName("wsname");
+        ws.setOwner("id-2314");
+        ws.setDefaultEnvName("env1");
+        ws.setEnvironments(Collections.singletonMap("env1", dto.createDto(EnvironmentDto.class).withName("env1")));
+        ws.setStatus(WorkspaceStatus.RUNNING);
+        ws.setCommands(Collections.singletonList(
+                dto.createDto(CommandDto.class).withName("MCI").withType("mvn").withCommandLine("clean install")));
+
+        UsersWorkspaceImpl usersWorkspace = ws.build();
+        when(workspaceManager.getWorkspace(eq(wsId))).thenReturn(usersWorkspace);
+
+        // when
+        Response response = given().auth().basic(JettyHttpServer.ADMIN_USER_NAME, JettyHttpServer.ADMIN_USER_PASSWORD).
```

Comment:
Line must never end with a DOT `.` see [doc](https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#for-dot--separator-break-always-comes-before-the-symbol)